src/fontcache.cpp

```diff
@@ -1056,9 +1056,301 @@ static void LoadWin32Font(FontSize fs)
 	new Win32FontCache(fs, logfont, settings->size);
 }
 
+#elif defined(WITH_COCOA)
+
+#include "os/macosx/macos.h"
+#include <CoreFoundation/CoreFoundation.h>
+#include <cmath>
+
+class CoreTextFontCache : public TrueTypeFontCache {
+	CFAutoRelease<CTFontDescriptorRef> font_desc; ///< Font descriptor exlcuding font size.
+	CFAutoRelease<CTFontRef> font;                ///< CoreText font handle.
+
+	std::string font_name;                        ///< Cached font name.
+
+	void SetFontSize(int pixels);
+	const Sprite *InternalGetGlyph(GlyphID key, bool aa) override;
+	const void *InternalGetFontTable(uint32 tag, size_t &length) override;
+public:
+	CoreTextFontCache(FontSize fs, CFAutoRelease<CTFontDescriptorRef> &&font, int pixels);
+	~CoreTextFontCache() {}
+
+	void ClearFontCache() override;
+	GlyphID MapCharToGlyph(WChar key) override;
+	const char *GetFontName() override { return font_name.c_str(); }
+	bool IsBuiltInFont() override { return false; }
+	const void *GetOSHandle() override { return font.get(); }
+};
+
+CoreTextFontCache::CoreTextFontCache(FontSize fs, CFAutoRelease<CTFontDescriptorRef> &&font, int pixels) : TrueTypeFontCache(fs, pixels), font_desc(std::move(font))
+{
+	this->SetFontSize(pixels);
+}
+
+/**
+ * Reset cached glyphs.
+ */
+void CoreTextFontCache::ClearFontCache()
+{
+	/* GUI scaling might have changed, determine font size anew if it was automatically selected. */
+	if (this->font) this->SetFontSize(this->req_size);
+
+	this->TrueTypeFontCache::ClearFontCache();
+}
+
+void CoreTextFontCache::SetFontSize(int pixels)
+{
+	if (pixels == 0) {
+		/* Try to determine a good height based on the height recommended by the font. */
+		int scaled_height = ScaleFontTrad(_default_font_height[this->fs]);
+		pixels = scaled_height;
+
+		CFAutoRelease<CTFontRef> f(CTFontCreateWithFontDescriptor(this->font_desc.get(), 0.0f, nullptr));
+		if (f) {
```

Comment:
`f`? What variable is that? :D Can I strongly suggest to use something not 1 letter? :D